src/client.rs

```diff
@@ -19,8 +21,34 @@ use tokio_service::Service;
 type WireResponse<Resp, E> = Result<Result<Resp, WireError<E>>, DeserializeError>;
 type ResponseFuture<Req, Resp, E> = futures::Map<<BindClient<Req, Resp, E> as Service>::Future,
                                             fn(WireResponse<Resp, E>) -> Result<Resp, ::Error<E>>>;
-type BindClient<Req, Resp, E> =
-    <Proto<Req, Result<Resp, WireError<E>>> as ProtoBindClient<Multiplex, TcpStream>>::BindClient;
+type BindClient<Req, Resp, E> = <Proto<Req, Result<Resp, WireError<E>>> as
+                                ProtoBindClient<Multiplex, StreamType>>::BindClient;
+
+/// TLS-specific functionality
+#[cfg(feature = "tls")]
+pub mod tls {
+    use native_tls::TlsConnector;
+
+    /// TLS context
+    pub struct TlsClientContext {
+        /// Domain to connect to
+        pub domain: String,
+        /// TLS connector
+        pub tls_connector: TlsConnector,
+    }
+
+    impl TlsClientContext {
+        /// Try to make a new `TlsClientContext`, providing the domain the client will
+        /// connect to.
+        pub fn try_new<S: Into<String>>(domain: S)
+                                        -> Result<TlsClientContext, ::native_tls::Error> {
+            Ok(TlsClientContext {
+                domain: domain.into(),
+                tls_connector: TlsConnector::builder()?.build()?,
```

Comment:
What else can the builder configure? Should we expose a builder for the inner builder? Should there be a separate `fn from_builder(builder: TlsConnectorBuilder) -> Result<Self, ::native_tls::Error>`?